src/core/strong_typedef_type.hpp

```diff
@@ -178,4 +178,31 @@ namespace StrongType {
 	};
 }
 
+/**
+ * Implementation of std::hash for StrongType::Typedef.
+ *
+ * This specialization of std::hash allows hashing of StrongType::Typedef instances
+ * by leveraging the hash of the base type.
+ *
+ * Example Usage:
+ *   using MyType = StrongType::Typedef<int, struct MyTypeTag>;
+ *   std::unordered_map<MyType, std::string> my_map;
+ *
+ * @tparam TBaseType The underlying type of the StrongType::Typedef.
+ * @tparam TTag A unique tag to distinguish different StrongType::Typedef types with the same base type.
+ * @tparam TProperties Additional properties for the StrongType::Typedef.
+ */
+template <typename TBaseType, typename TTag, typename... TProperties>
+struct std::hash<StrongType::Typedef<TBaseType, TTag, TProperties...>> {
```

Comment:
Practically `TTag` can be removed here and on line 203. You're only interested in the `TBaseType` and the tag can also be considered a property.